modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -475,7 +483,7 @@ private ReadOnlyTransactionImpl beginReadOnlyTransaction(
     }
 
     private long roExpirationPhysicalTimeFor(HybridTimestamp beginTimestamp, InternalTxOptions options) {
-        long effectiveTimeoutMillis = options.timeoutMillis() == 0 ? defaultTransactionTimeoutMillis() : options.timeoutMillis();
+        long effectiveTimeoutMillis = options.timeoutMillis() == 0 ? defaultReadOnlyTransactionTimeoutMillis() : options.timeoutMillis();
```

Comment:
This calculation is repeated at least twice. Let's extract it to a method

Reply:
The next ticket will change this, don't you mind to leave this as is?

Reply:
I'm ok with this if this will not be forgotten :)